@trycourier/courier-js/src/socket/courier-socket.ts

```diff
@@ -63,6 +63,15 @@ export abstract class CourierSocket {
   /** The timeout ID for the current connectionretry attempt, reset when we attempt to connect. */
   private retryTimeoutId: number | null = null;
 
+  /**
+   * Flag indicating the application initiated a {@link CourierSocket#close} call.
+   *
+   * An application-initiated close may look like an abnormal closure (code 1006)
+   * if it occurs before the connection is established. We differentiate to
+   * prevent retrying the connection when the socket is closed intentionally.
+   */
+  private closeRequested: boolean = false;
```

Comment:
I don't love keeping track of this state here, but we need a way to differentiate a real network error from one induced by calling `close()` and creating a `CourierSocketState` seemed like it'd have too much overlap with the native `WebSocket.ReadyState`.